### PR TITLE
[Serve] Make the checkpoint and recover only from GCS

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

Make the GCS as only way to recover. (Not well tested yet for external storage)

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by simon-mo** [comment]
this is also mentioned in documentation. Can you take care of that?

**Review by edoakes — COMMENTED** [review]
I think we can also get rid of most of the code under `serve/storage`.

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -643,17 +640,8 @@ def build(target: Union[ClassNode, FunctionNode]) -> Application:
 
 
 def _check_http_and_checkpoint_options(
```
Comment (simon-mo) [inline python/ray/serve/api.py:642]: naming

**Review by simon-mo — APPROVED** [review]
also feel free to remove s3 and other stores in storage folder! since they are now no longer used.

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
@sihanwang41 DCO build failed. Need to signoff on your commits. Use this to squash the changes here into one signed-off commit:
```
# From your branch.
git reset $(git merge-base master $(git branch --show-current))
git commit -s -m <your message>
git push -f
```

**Pull request merged**